### PR TITLE
Better predicate names and error handling

### PR DESCRIPTION
**Pull request by Esaron**
This is a followup to https://github.com/betterup/wisper-activejob-broadcaster/pull/3 with slightly less risk. See https://github.com/betterup/betterup-app/pull/26435 for more context.

### CODE REVIEW

**Review by f1sherman — APPROVED** [review]

File: lib/wisper/activejob/broadcaster/version.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module Wisper
   module Activejob
     module Broadcaster
-      VERSION = '0.2.1'.freeze
+      VERSION = '0.3.1'.freeze
```
Comment (wireframe) [inline lib/wisper/activejob/broadcaster/version.rb:4]: When bumping a "minor" version number, the "patch" version number typically resets to zero.

File: lib/wisper/activejob/broadcaster.rb [hunk 2]
```diff
@@ -13,9 +13,39 @@ def broadcast(subscriber, _publisher, event, args)
...
+
+        def args_info(args)
+          return if args.empty?
+
+          args.map do |arg|
+            arg_string = name(arg)
+            arg_string += ": #{arg.inspect}" if [Numeric, Array, Hash, String].any? {|klass| arg.is_a?(klass) }
+            arg_string
+          end.join(', ')
+        end
```
Comment (wireframe) [inline lib/wisper/activejob/broadcaster.rb:48]: I'm missing the context for this change.  What is the purpose of rescuing and rethrowing the error here?  Isn't this swallowing the original error and losing the context?

File: lib/wisper/activejob/broadcaster/version.rb (hunk 1 above)
Comment (Esaron) [reply lib/wisper/activejob/broadcaster/version.rb:4]: 🤦  Thanks!

File: lib/wisper/activejob/broadcaster.rb (hunk 2 above)
Comment (Esaron) [reply lib/wisper/activejob/broadcaster.rb:48]: This is just to make sure we don't fail without a way to rerun the job we would have previously fired off. Today, if we create a job for a listener method and it fails, sidekiq retry logic protects us. With the previous change to this gem, we'd never create the job and we'd lose the information needed to recreate it.

The `cause` of the new `DispatchError` should be the error we're rescuing. I'll have to double check that that's good enough.

Comment (Esaron) [reply lib/wisper/activejob/broadcaster.rb:48]:
```
irb(main):083:1* def foo
irb(main):084:2*   begin
irb(main):085:2*     raise StandardError.new('test')
irb(main):086:2*   rescue StandardError => e
irb(main):087:2*     raise NoMethodError.new('othertest')
irb(main):088:1*   end
irb(main):089:0> end
=> :foo
irb(main):090:0> foo
(irb):87:in `rescue in foo': othertest (NoMethodError)
	from (irb):83:in `foo'
	from (irb):90:in `<main>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `<main>'
(irb):85:in `foo': test (StandardError)
	from (irb):90:in `<main>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/irb-1.3.5/exe/irb:11:in `<top (required)>'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `load'
	from /Users/jmarshall/.rbenv/versions/3.0.2/bin/irb:23:in `<main>'
irb(main):091:1* begin
irb(main):092:1*   foo
irb(main):093:1* rescue StandardError => e
irb(main):094:1*   puts e.inspect
irb(main):095:1*   puts e.cause.inspect
irb(main):096:0> end
#<NoMethodError: othertest>
#<StandardError: test>
=> nil
```

Comment (Esaron) [reply lib/wisper/activejob/broadcaster.rb:48]: I'm pretty sure the whole exception chain gets logged, so we shouldn't be losing any info.

**Review by wireframe — APPROVED** [review]

**Comment by wireframe** [comment]
Released as part of 0.3.0!

**Pull request merged**